src/org/zaproxy/zap/extension/scripts/ExtensionScriptsUI.java

```diff
@@ -172,6 +183,24 @@ public boolean canUnload() {
     	return true;
     }
 	
+	@Override
+	public void optionsLoaded() {
+		// Run any enabled scripts
+		for (ScriptWrapper script : this.getExtScript().getScripts(extScriptType)) {
+			if (script.isEnabled()) {
+				// Run the install 
+				ExtenderScript ec;
+				try {
+					ec = extScript.getInterface(script, ExtenderScript.class);
+					ec.install(getExtensionScriptHelper());
+					this.installedExtenderScripts.add(script.getName());

```

Comment:
- I'd suggest using a map (`ScriptWrapper` to `ExtenderScript`?) instead of just tracking if they are installed, that ensures it is executed the correct/expected uninstallation code, in case changes are done to the script after being installed. Also, that means the state of the script is preserved so the registration of the components is no longer needed (they can be kept in script's variables).
- The error state of the script should be reset after this statement.